tests/test_hexary_trie.py

```diff
@@ -289,6 +289,84 @@ def test_hexary_trie_batch_save_drops_last_root_data_when_pruning():
     assert encode_hex(old_root_hash) in str(excinfo.value)
 
 
+def test_squash_changes_can_still_access_underlying_deleted_data():
+    db = {}
+    trie = HexaryTrie(db, prune=True)
+    trie.set(b'what floats on water?', b'very small rocks')
+    old_root_hash = trie.root_hash
+
+    with trie.squash_changes() as memory_trie:
+        memory_trie.set(b'what floats on water?', b'a duck')
+
+        # change to a root hash that the memory trie doesn't have anymore
+        new_root_hash = memory_trie.root_hash
+        memory_trie.root_hash = old_root_hash
+
+        assert memory_trie[b'what floats on water?'] == b'very small rocks'
+
+        # change the root hash back because yikes it's confused now
+        memory_trie.root_hash = new_root_hash
+
+
+def test_squash_changes_raises_correct_error_on_new_deleted_data():
+    db = {}
+    trie = HexaryTrie(db, prune=True)
+    trie.set(b'what floats on water?', b'very small rocks')
+
+    with trie.squash_changes() as memory_trie:
+        memory_trie.set(b'what floats on water?', b'a duck')
+        middle_root_hash = memory_trie.root_hash
+
+        memory_trie.set(b'what floats on water?', b'ooooohh')
+        new_root_hash = memory_trie.root_hash
+
+        # change to a root hash that the memory trie doesn't have anymore
+        memory_trie.root_hash = middle_root_hash
+
+        with pytest.raises(MissingTrieNode):
+            memory_trie[b'what floats on water?']
+
+        # change the root hash back because yikes it's confused now
```

Comment:
You've got this message in two places and it isn't clear to me what it is trying to convey.

Reply:
Looking back, I realize that it was just me being bothered with the idea of manually tweaking the root in the middle of the squash. It doesn't actually hurt anything to leave it modified. I removed the comment and the the `root_hash` reset line.